src/operator/linalg_impl.h

```diff
@@ -985,42 +985,85 @@ template<> inline \
 void linalg_syevd<cpu, DType>(const Tensor<cpu, 2, DType>& A, \
                               const Tensor<cpu, 1, DType>& L, \
                               const Tensor<cpu, 1, DType>& work, \
-                              const Tensor<cpu, 1, int>& iwork, \
                               Stream<cpu> *s) { \
   check_syevd(A, L); \
+  int liwork(0); \
+  MXNET_LAPACK_##fname(MXNET_LAPACK_ROW_MAJOR, 'L', A.size(0), \
```

Comment:
I don't like this. You are doing a second workspace query call here, because you don't know how many entries the initial DType part of work should have. But really, the workspace query has already been done.

There must be a better solution. Either pass lwork as additional argument, which is just not used in the GPU case, or have the first entry of work be set to lwork in the CPU case. I prefer the former.